subsys/net/ip/net_if.c

```diff
@@ -3154,6 +3154,8 @@ void net_if_post_init(void)
 
 	/* After TX is running, attempt to bring the interface up */
 	for (iface = __net_if_start; iface != __net_if_end; iface++) {
-		net_if_up(iface);
+		if (!net_if_flag_is_set(iface, NET_IF_NO_AUTO_START)) {
+			net_if_up(iface);
```

Comment:
Following up discussion in https://github.com/zephyrproject-rtos/zephyr/pull/15480/commits/dc115e6d20ca711c424cfc2fc90cfc9327349d0b#r275690049 , indeed, what if we make calling net_if_up(iface) a driver responsibility?

Reply:
That is exactly what we are doing here by introducing this flag. I did not wanted to change the current functionality thou, so by default we still take the interface up. This could be changed in the future.

Reply:
I meant completely removing this "for" loop and making driver init routine call net_if_up(). I feel this this probably will lead to further sync issues between different network stack parts (a-la https://github.com/zephyrproject-rtos/zephyr/issues/13991). So, let me approve this.

Reply:
driver should not drive the iface state (moreover when there can be more than 1 iface per-device on VLAN for instance). The issue is really about setting carrier off/on properly at init time.

Reply:
@tbursztyka How would a modem driver let the network stack know that connection was down?  Seems like drivers should have some input to their own interface state?

Reply:
@mike-scott use `net_if_carrier_down()` or `net_if_down()` when the low level connection goes down, and then `net_if_up()` when it goes up. For example Ethernet has its own `net_eth_carrier_on()` and `net_eth_carrier_off()` that basically calls those functions.

Reply:
AH makes sense.  Driver manages the carrier status.  Thank you for the explanation.